app/src/main/java/com/github/dedis/student20_pop/MainActivity.java

```diff
@@ -66,20 +71,41 @@ public void onClick(View view) {
                 String name = ((EditText) findViewById(R.id.entry_box_launch)).getText().toString();
-                Person organizer = ((PoPApplication) getApplication()).getPerson();
+                final PoPApplication app = ((PoPApplication) getApplication());
                 // Creating the LAO and adding it to the organizer's LAO
-                Lao lao = new Lao(name, new Date(), organizer.getId());
-                organizer = organizer.setLaos(Collections.singletonList(lao.getId()));
+                Lao lao = new Lao(name, new Date(), app.getPerson().getId());
                 // Store the private key of the organizer
-                if (PrivateInfoStorage.storeData(this, organizer.getId(), organizer.getAuthentication()))
+                //TODO Move it into app onCreate()
+                if (PrivateInfoStorage.storeData(this, app.getPerson().getId(), app.getPerson().getAuthentication()))
                     Log.d(TAG, "Stored private key of organizer");
-                // TODO: send LAO and organizer information to backend
-                // Set LAO and organizer information locally
-                ((PoPApplication) getApplication()).setPerson(organizer);
-                ((PoPApplication) getApplication()).setLaos(Collections.singletonList(lao));
-                // Start the Organizer Activity (user is considered an organizer)
-                Intent intent = new Intent(this, OrganizerActivity.class);
-                startActivity(intent);
+
+                CompletableFuture.supplyAsync(() -> {
+                    try {
+                        //TODO Get URL dynamically
+                        return PoPClientEndpoint.connectToServer(URI.create("ws://10.0.2.2:2020/"), app.getPerson());
+                    } catch (DeploymentException e) {
+                        e.printStackTrace();
+                        return null;
+                    }
+                }).thenCompose(p -> {
+                    if(p != null)
+                        return p.createLoa(lao.getName(), lao.getTime(), lao.getTime(), app.getPerson().getId());
+                    else
+                        return null;
+                }).whenComplete((errCode, t) -> {
+                    if(t != null)
+                        //TODO Show toast
+                        Log.e(TAG, "Error while creating the lao", t);
+                    else {
+                        Person organizer = app.getPerson().setLaos(Collections.singletonList(lao.getId()));
+                        // Set LAO and organizer information locally
+                        ((PoPApplication) getApplication()).setPerson(organizer);
+                        ((PoPApplication) getApplication()).setLaos(Collections.singletonList(lao));
+                        // Start the Organizer Activity (user is considered an organizer)
+                        Intent intent = new Intent(this, OrganizerActivity.class);
+                        startActivity(intent);
+                    }
```

Comment:
Maybe you can add a TODO to show a Toast so the user knows the LAO was crated successfully

Reply:
Done !